js/pro/krakenfutures.js

```diff
@@ -0,0 +1,239 @@
+        const market = this.market (product_id);
+        const symbol = market['symbol'];
+        const messageHash = name + ':' + symbol;
+        const timestamp = this.milliseconds ();
+        const result = {
+            'symbol': symbol,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': undefined,
+            'low': undefined,
+            'bid': this.safeFloat (message, 'bid'),
+            'bidVolume': this.safeFloat (message, 'bid_size'),
+            'ask': this.safeFloat (message, 'ask'),
+            'askVolume': this.safeFloat (message, 'ask_size'),
+            'vwap': undefined,
+            'open': undefined,
+            'close': undefined,
+            'last': this.safeFloat (message, 'last'),
+            'previousClose': undefined,
+            'change': undefined,
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': this.safeFloat (message, 'volume'),
+            'quoteVolume': this.safeFloat (message, 'volumeQuote'),
+            'info': message,
+        };
+        this.tickers[symbol] = result;
+        client.resolve (result, messageHash);
+    }
+
+    async watchPublic (name, symbol, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const messageHash = name + ':' + market['symbol'];
+        const url = this.urls['api']['ws']['public'];
+        const subscribe = {
+            'event': 'subscribe',
+            'feed': name,
+            'product_ids': [
+                market['id'],
+            ],
+        };
+        const request = this.deepExtend (subscribe, params);
+        return await this.watch (url, messageHash, request, messageHash);
+    }
+
+    async watchTicker (symbol, params = {}) {
+        /**
+         * @method
+         * @name krakenfutures#watchTicker
+         * @description watches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+         * @param {string} symbol unified symbol of the market to fetch the ticker for
+         * @param {object} params extra parameters specific to the kraken api endpoint
+         * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        return await this.watchPublic ('ticker', symbol, params);
+    }
+
+    async watchHeartbeat (params = {}) {
+        await this.loadMarkets ();
+        const event = 'heartbeat';
+        const url = this.urls['api']['ws']['public'];
+        return await this.watch (url, event);
+    }
+
+    handleHeartbeat (client, message) {
+        //
+        // every second (approx) if no other updates are sent
+        //
+        //     { "event": "heartbeat" }
+        //
+        const event = this.safeString (message, 'event');
+        client.resolve (message, event);
+    }
+
+    async authenticate (params = {}) {
```

Comment:
Isn't this a leftover from another exchange? (kraken I think). If so we should remove it